var/spack/repos/builtin/packages/abinit/package.py

```diff
@@ -0,0 +1,175 @@
+    url      = "http://ftp.abinit.org/abinit-8.0.8b.tar.gz"
+
+    # Versions before 8.0.8b are not supported.
+    version("8.0.8b", "abc9e303bfa7f9f43f95598f87d84d5d")
+
+    variant('mpi', default=True,
+            description='Builds with MPI support. Requires MPI2+')
+    variant('openmp', default=False,
+            description='Enables OpenMP threads. Use threaded FFTW3')
+    variant('scalapack', default=False,
+            description='Enables scalapack support. Requires MPI')
+    # variant('elpa', default=False,
+    # description='Uses elpa instead of scalapack. Requires MPI')
+
+    # TODO: To be tested.
+    # It was working before the last `git pull` but now all tests crash.
+    # For the time being, the default is netcdf3 and the internal fallbacks
+    variant('hdf5', default=False,
+            description='Enables HDF5+Netcdf4 with MPI. WARNING: experimental')
+
+    # Add dependencies
+    # currently one cannot forward options to virtual packages, see #1712.
+    # depends_on("blas", when="~openmp")
+    # depends_on("blas+openmp", when="+openmp")
+    depends_on('blas')
+    depends_on("lapack")
+
+    # Require MPI2+
+    depends_on("mpi@2:", when="+mpi")
+
+    depends_on("scalapack", when="+scalapack+mpi")
+    # depends_on("elpa", when="+elpa+mpi~openmp")
+    # depends_on("elpa+openmp", when="+elpa+mpi+openmp")
+
+    depends_on("fftw+float", when="~openmp")
+    depends_on("fftw+float+openmp", when="+openmp")
+
+    depends_on("netcdf-fortran", when="+hdf5")
+    depends_on("hdf5+mpi", when='+mpi+hdf5')  # required for NetCDF-4 support
+
+    # pin libxc version
+    depends_on("libxc@2.2.1")
+
+    def validate(self, spec):
+        """
+        Checks if incompatible variants have been activated at the same time
+
+        :param spec: spec of the package
+        :raises RuntimeError: in case of inconsistencies
+        """
+        error = 'you cannot ask for \'+{variant}\' when \'+mpi\' is not active'
+
+        if '+scalapack' in spec and '~mpi' in spec:
+            raise RuntimeError(error.format(variant='scalapack'))
+
+        if '+elpa' in spec and ('~mpi' in spec or '~scalapack' in spec):
+            raise RuntimeError(error.format(variant='elpa'))
+
+    def install(self, spec, prefix):
+        self.validate(spec)
+
+        options = ['--prefix=%s' % prefix]
+        oapp = options.append
+
+        if '+mpi' in spec:
+            # MPI version:
+            # let the configure script auto-detect MPI support from mpi_prefix
+            oapp("--with-mpi-prefix=%s" % spec["mpi"].prefix)
+            oapp("--enable-mpi=yes")
+            oapp("--enable-mpi-io=yes")
+
+        # Activate OpenMP in Abinit Fortran code.
+        if '+openmp' in spec:
+            oapp('--enable-openmp=yes')
+
+        # BLAS/LAPACK
+        if '+scalapack' in spec:
+            oapp("--with-linalg-flavor=custom+scalapack")
+            linalg = (spec['lapack'].lapack_libs + spec['blas'].blas_libs +
```

Comment:
@gmatteo sourceforge often has download issues. Most probably the website is down and it displays an error page instead of giving Spack an archive. You can hack an alternative download source in `expat` package.